eclipse-theia/theia
# [remote] Support download of native dependencies

#### What it does
This is a part of the Remote SSH feature #12390.
This PR is supposed to solve the problem of needing different native dependencies when copying the backend to a remote machine with a different OS or cpu-architecture.
As discussed in the theia-dev meeting on 6.6.23 these nativ dependencies should be included as assets in the respective theia release.

#### How to test
Its kind of hard to test since the rest of the feature is not yet integrated or used by anything.

The Github Actions workflow file is also deployed [here](https://github.com/jonah-iden/theia-native-dependencies/actions/workflows/node.js.yml) for testing.
For testing the download contributions, it's probably easiest to create a small `BackendApplicationContribution` in the `remote` package and execute the `NativeDependencyDownloadService` `downloadDependencies` in `onStart()` in there like this:
```Typescript
@injectable()
export class TestBackendContrib implements BackendApplicationContribution {
    @inject(DependencyDownloadService) service: DependencyDownloadService;
    onStart(): void {
        this.service.downloadDependencies(`${process.platform}-${process.arch}`);
    }
}
```
 
After that at least for rigrep (all other dependencies are not yet deployed so yet so it kind of download the website). Through debugging you could see that they are downloaded. or maybe writing the to the file system
#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
